blocks/edit/prose/plugins/keyHandlers.js

```diff
@@ -82,6 +111,12 @@ export function getURLInputRulesPlugin() {
   return inputRules({ rules: [getURLInputRule()] });
 }
 
+// Returns an inputRules plugin for list formatting (bullet and ordered lists)
+export function getListInputRulesPlugin(schema) {
+  const rules = [getBulletListInputRule(schema), getOrderedListInputRule(schema)].filter(Boolean);
```

Comment:
I don't think there's ever a time where `schema.nodes` will NOT have `bullet_list` and `ordered_list`, so it should be safe to assume that they will be present.

I think that all of the listInputRule code can be consolidated into this function:
```
export function getListInputRulesPlugin(schema) {
  const rules = [];
  const olSchema = schema.nodes['ordered_list'];
  if (olSchema) rules.push(olSchema);
  const blSchema = schema.nodes['bullet_list'];
  if (blSchema) rules.push(blSchema); 
  return inputRules({ rules });
}
```

Reply:
makes sense to me, but if `schema.nodes.bullet_list` and `schema.nodes.ordered_list` will always be defined, then we don't need the `if` statements at all right?

Assuming that's true, I simplified it to just:

```
export function getListInputRulesPlugin(schema) {
  const rules = [
    wrappingInputRule(/^\s*1[.)]\s$/, schema.nodes.ordered_list),
    wrappingInputRule(/^\s*([-*])\s$/, schema.nodes.bullet_list),
  ];
  return inputRules({ rules });
}
```

If you think the ifs are needed, let me know and happy to update to the suggestion your provided.